packages/ckeditor5-engine/tests/model/utils/insertcontent.js

```diff
@@ -1299,128 +1302,1629 @@ describe( 'DataController utils', () => {
+					}, [ 0 ] );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph>Bar</paragraph><paragraph>[]foo</paragraph>' );
+
+					expect( expectedMarker ).to.exist;
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 1 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 1 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageInline at the beginning of paragraph', () => {
+					// <paragraph>[]foo</paragraph>
+					// {}<imageInline></imageInline>
+					//
+					// <paragraph>{}<imageInline></imageInline>foo</paragraph>
+					setData( model, '<paragraph>[]foo</paragraph>' );
+
+					insertHelper( '<imageInline></imageInline>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph><imageInline></imageInline>[]foo</paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 0 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 0 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageInline in the middle of paragraph', () => {
+					// <paragraph>f[]oo</paragraph>
+					// {}<imageInline></imageInline>
+					//
+					// <paragraph>f{}<imageInline></imageInline>oo</paragraph>
+					setData( model, '<paragraph>f[]oo</paragraph>' );
+
+					insertHelper( '<imageInline></imageInline>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph>f<imageInline></imageInline>[]oo</paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 1 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 1 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageInline at the end of paragraph', () => {
+					// <paragraph>foo[]</paragraph>
+					// {}<imageInline></imageInline>
+					//
+					// <paragraph>foo{}<imageInline></imageInline></paragraph>
+					setData( model, '<paragraph>foo[]</paragraph>' );
+
+					insertHelper( '<imageInline></imageInline>', {
+						'marker-a': { start: [ 0 ], end: [ 0 ] }
+					} );
+
+					const expectedMarker = model.markers.get( 'marker-a' );
+
+					expect( getData( model ) ).to.equal( '<paragraph>foo<imageInline></imageInline>[]</paragraph>' );
+
+					expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 0, 3 ] );
+					expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 0, 3 ] );
+				} );
+
+				it( 'should create collapsed marker after inserting imageInline next to paragraph', () => {
+					// <paragraph>foo</paragraph>[]
+					// {}<imageInline></imageInline>
+					//
+					// <paragraph>foo[]</paragraph>{<paragraph><imageInline></imageInline></paragraph>}
```

Comment:
It should either be before `<paragraph>` or before `<imageInline>` but not like this. AFAICS, the expect is correct but the comment is wrong.

Reply:
The question here is what behavior we see as correct regarding the positioning of the collapsed marker, i.e., should it be at the beginning of inserted content, or should it stay next to element that it was in the original content.

In other words, what is correct: `{}<paragraph><imageInline>` or `<paragraph>{}<imageInline>`. I believe the second one and we can see what will happen to this test after we improve the solution.

Reply:
I updated this test and the answer is the first one: `{}<paragraph><imageInline>`

Reply:
:+1:, I expected this to end up as `{}<paragraph>`, what I meant by "correct" is what we want to achieve / what is the user's expected behavior.